asmcli/lib/util.sh

```diff
@@ -694,14 +695,35 @@ init_meshconfig_curl() {
 }
 
 get_gke_release_channel() {
+  if [[ -n "${GKE_RELEASE_CHANNEL}" ]]; then echo "${GKE_RELEASE_CHANNEL}"; return; fi
+
   local PROJECT_ID; PROJECT_ID="$(context_get-option "PROJECT_ID")"
   local CLUSTER_NAME; CLUSTER_NAME="$(context_get-option "CLUSTER_NAME")"
   local CLUSTER_LOCATION; CLUSTER_LOCATION="$(context_get-option "CLUSTER_LOCATION")"
-  gcloud container clusters describe \
+  GKE_RELEASE_CHANNEL="$(gcloud container clusters describe \
     --project="${PROJECT_ID}" \
     --region "${CLUSTER_LOCATION}" \
     "${CLUSTER_NAME}" \
-    --format="value(releaseChannel.channel)"
+    --format="value(releaseChannel.channel)")"
+  readonly GKE_RELEASE_CHANNEL; echo "${GKE_RELEASE_CHANNEL}"
+}
+
+channels_match() {
+  [[ "$(get_cr_channel)" == "$(get_gke_release_channel)" ]]
+}
+
+use_fleet_api() {
+  # At first, must be explicitly asked for as well as meeting criteria.
+  local EXPLICIT_FLEET_API; EXPLICIT_FLEET_API="$(context_get-option "EXPLICIT_FLEET_API")"
+  if [[ "${EXPLICIT_FLEET_API}" -ne 1 ]]; then
+    false
+    return
+  fi
+
+  local CA; CA="$(context_get-option "CA")"
+  if channels_match && [[ "$(get_cr_channel)" != "stable" ]] && [[ "${CA}" != *cas ]]; return; fi
+
+  warn "Fleet API requested, but requirements not met so using old method."
```

Comment:
Eventually we will. I don't want customers to do something like this:

1) discover the fleet api switch and turn it on
2) observe that it works
3) later, we change the criteria to be more strict
4) now automated provisioning that used to work has broken

So I'm opting to be more permissive at first until we get more confidence in what we need to know to be confident that the fleet API is safe to use.

Reply:
I don't really like that. To me it seems worse if automated provisioning that used to work is now doing something different. At least if it errors out they know they have to fix something!

But you know more about the desired asmcli behavior so I'll let you make the call.

Reply:
On one hand I agree, but also until this is on a release branch there's no expectation of consistent behavior anyways.

I think you convinced me and I'll probably change it in the next PR, but I'm going to selfishly resolve this now so that I don't have to wait for tests to pass again before submitting. If I forget to change it later then I'll accept my official demerits.

Reply:
I of course said that and then noticed that the tests didn't pass anyways, so I'll change it :P